src/com/owncloud/android/media/MediaService.java

```diff
@@ -250,11 +251,25 @@ public int onStartCommand(Intent intent, int flags, int startId) {
             
         } else if (action.equals(ACTION_STOP_ALL)) {
             processStopRequest(true);
+        } else if(action.equals(ACTION_STOP_FILE)) {
+            processStopFileRequest(intent);
         }
 
         return START_NOT_STICKY; // don't want it to restart in case it's killed.
     }
 
+    private void processStopFileRequest(Intent intent) {
+        OCFile file = intent.getExtras().getParcelable(EXTRA_FILE);
+        try {
+            if(file.equals(mFile)) {
+                processStopRequest(true);
+            }
+        }
+        catch (NullPointerException e) {
```

Comment:
In general, ```NullPointerException```s shouldn't be cached - the input of the block should be tested to prevent it occurs.